src/app/authentication/screens/login-screen.component.tsx

```diff
@@ -84,51 +77,74 @@ export const LoginText = styled.Text`
-    margin-top: ${(props) => props.theme.spacer}px;
+    margin-top: ${(props) => props.theme.spacer * 2.5}px;
+    font-family: ${(props) => props.theme.fontFamily.bold};
+`;
+export const BackgroundImage = styled.ImageBackground`
+    width: ${DEVICE_SIZE.width};
+    height: ${DEVICE_SIZE.height + 2};
+    position: absolute;
+`;
+export const SignUpWrapper = styled.TouchableOpacity`
+    width: ${DEVICE_SIZE.width * 0.3};
+    margin-left: ${DEVICE_SIZE.width * 0.27};
 `;
 
-export const LoginScreen: React.FC<LoginScreenProps> = () => {
+export const validateEmail: (string) => boolean = (email: string) => {
+    const re = /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/;
+    return re.test(String(email).toLowerCase());
+};
+
+export const LoginScreen: React.FC<LoginScreenProps> = (props: LoginScreenProps) => {
     const dispatch = useDispatch();
     const themeKey = useSelector(getTheme);
-    const theme = useMemo(() => themesCollection[themeKey] as DefaultTheme, [themeKey]);
-
-    const handleLogin = useCallback(() => {
-        dispatch(
-            LOGIN.TRIGGER({
-                username: 'vlad',
-                password: '123',
-            })
-        );
-    }, [dispatch]);
+    const theme = useTheme();
+    const [email, setEmail] = useState('');
+    const [password, setPassword] = useState('');
+    const [validation, setValidation] = useState(true);
 
-    const handleTheme = useCallback(() => {
-        dispatch(CHANGE_THEME({ theme: ThemesEnum.LIGHT }));
-    }, [dispatch]);
+    const handleLogin = () => {
+        dispatch(LOGIN.TRIGGER({ email: email, password: password }));
+    };
 
     return (
-        <Container>
-            {theme.colors.main === 'black' ? (
-                <Logo source={require('../../../assets/dark-logo.jpg')} />
-            ) : (
-                <Logo source={require('../../../assets/light-logo.jpg')} />
-            )}
-            <Title>Splash</Title>
-            <InputArea>
-                <EmailText>Email</EmailText>
-                <Input />
-                <PasswordText>Password</PasswordText>
-                <Input secureTextEntry={true} />
-                <LoginButton activeOpacity={0.5} onPress={handleLogin}>
-                    <LoginText>Sign in</LoginText>
-                </LoginButton>
-                <TouchableOpacity activeOpacity={0.5} onPress={handleTheme}>
-                    <SignUpText>Sign up</SignUpText>
-                </TouchableOpacity>
-            </InputArea>
-        </Container>
+        <AvoidingContainer>
+            <KeyboardAvoidingView
+                contentContainerStyle={{ flex: 1 }}
+                style={{ flex: 1 }}
+                behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
+            >
+                {themeKey === ThemesEnum.DARK || ThemesEnum.JAPANESE ? (
+                    <BackgroundImage source={require('../../../assets/background.jpg')} />
```

Comment:
We have ImageBackground component in React Native. Refactor

Reply:
> 

this is it

Reply:
You can add ternar operator on source prop